### ISSUE TITLE
Use push events to poll for new job runs

### ISSUE THREAD

**Issue by simmel** [nedbat/watchgha#9]
Awesome tool!

I wanted to yak shave something similar but I just wanted to run in another terminal to see that my CI/CD got a green bar, not start a process after every time I've pushed and Github Actions has started the job (which can take some time sometimes).

My idea was:
Poll https://docs.github.com/en/rest/reference/activity#list-repository-events and look for [push events](https://docs.github.com/en/developers/webhooks-and-events/events/github-event-types#pushevent). I know it says `This API is not built to serve real-time use cases. Depending on the time of day, event latency can be anywhere from 30s to 6h.` but that's just a warning, I suspect it's pretty much always fast except if Githubs "bus" is very backlogged.

My additional idea was to [list workflow](https://docs.github.com/en/rest/reference/actions#list-workflow-runs-for-a-repository) and then from the [workflow runs](https://docs.github.com/en/rest/reference/actions#list-jobs-for-a-workflow-run) display the [job logs](https://docs.github.com/en/rest/reference/actions#download-job-logs-for-a-workflow-run) but it turns out you can't stream them; the logs are only available when the job is done.

Yay or nay?

**Comment by nedbat** [comment]
Hi, sorry I haven't commented on this yet.  I don't understand how you would use watchgha after this change.  It would repeatedly poll the repo events, and when it saw a push event, it would start showing those actions?

**Comment by simmel** [comment]
Exactly. So think `tail -F` but for a repos actions. So `watchgha -F simmel/somerepo` and everytime I commit and a build action starts `watchgha` starts so show the status.

**Comment by simmel** [comment]
An extension later on when an action run (what is the real term?) is complete and "green" you should "minimize" the run to one line with a green check. So just how the topmost line in the gif in the README and add a green check after.
